src/ssh/cockpitsshrelay.c

```diff
@@ -1374,12 +1375,15 @@ cockpit_ssh_connect (CockpitSshData *data,
     }
 
   g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_HOST, host) == 0);
-#if LIBSSH_085
-    g_warn_if_fail (ssh_options_parse_config (data->session, NULL) == 0);
-#endif
-  g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_USER, data->username) == 0);
   if (port != 0)
     g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_PORT, &port) == 0);
+#if LIBSSH_085
+  g_warn_if_fail (ssh_options_parse_config (data->session, NULL) == 0);
+#endif
+
+  /* If the ssh config doesn't specify User, use the username from the auth message */
+  if (ssh_options_get (data->session, SSH_OPTIONS_USER, &username) != 0)
```

Comment:
You can't use `g_autofree` with a string returned from `ssh_options_get()`.  Sorry bebs.